### PR TITLE
Documenting the delta between EDB system catalogs and information available in PostgreSQL

### PR DESCRIPTION
**Pull request by dwicinas**
Added files for:

pg_attribute
pg_authid
pg_class
pg_collation
pg_database
pg_namespace
pg_partitioned_table
pg_proc
pg_subscription
pg_synonym
pg_trigger
pg_type
pg_user

Addresses https://enterprisedb.atlassian.net/browse/DF-368

### CODE REVIEW

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_attribute.mdx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: "PG_ATTRIBUTE"
+---
+
+The `PG_ATTRIBUTE` view provides information about table columns. There is one `PG_ATTRIBUTE` row for every column in every table in the database. In addition to the information in the [PostgreSQL pg_attribute description](https://www.postgresql.org/docs/current/catalog-pg-attribute.html), EDB Postgres Advanced Server provides the following information:
+
+| Name              | Type      | Description                                                                            |
+| ----------------- | --------- | -------------------------------------------------------------------------------------- |
+| attishidden       | boolean   | Column included in SELECT * expansion                                                  |
```
Comment (nidhibhammar) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_attribute.mdx:9]:
```suggestion
| attishidden       | boolean   | Column in SELECT * expansion to be hidden                                               |
```

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_attribute.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: "PG_ATTRIBUTE"
+---
+
+The `PG_ATTRIBUTE` view provides information about table columns. There is one `PG_ATTRIBUTE` row for every column in every table in the database. In addition to the information in the [PostgreSQL pg_attribute description](https://www.postgresql.org/docs/current/catalog-pg-attribute.html), EDB Postgres Advanced Server provides the following information:
+
+| Name              | Type      | Description                                                                            |
+| ----------------- | --------- | -------------------------------------------------------------------------------------- |
+| attishidden       | boolean   | Column in SELECT * expansion to be hidden                                               |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_attribute.mdx:9]: Column included in SELECT * expansion

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_authid.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+| rolcreatedblink       | boolean                  | Specifies whether the role is allowed to create database links        |
+| rolcreatepublicdblink | boolean                  | Specifies whether the role is allowed to create public database links |
+| roldroppublicdblink   | boolean                  | Specifies whether the role is allowed to drop public database links   |
+| rolfailedlogins       | integer                  | Number of failed logins                                               |
+| rollockdate           | timestamp with time zone | Time when the account was locked                                      |
+| rolpasswordexpire     | timestamp with time zone | Password expiration time, if applicable, for Redwood accountability   |
+| rolpasswordsetat      | timestamp with time zone | Time when the password was set, if applicable                         |
+| rolpolicyexempt       | boolean                  | Specifies whether the role is allowed to bypass VPD policies          |
+| rolprofile            | oid                      | Object identifier for the role's profile                              |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_authid.mdx:18]: Oid                     oid;                    /* oid */
        NameData        rolname;                /* name of role */
        bool            rolsuper;               /* read this field via superuser() only! */
        bool            rolinherit;             /* inherit privileges from other roles? */
        bool            rolcreaterole;  /* allowed to create more roles? */
        bool            rolcreatedb;    /* allowed to create databases? */
        bool            rolcreatedblink;/* allowed to create databases link? */
        bool            rolcreatepublicdblink;/* allowed to create public databases link? */
        bool            roldroppublicdblink;/* allowed to drop public databases link? */
        bool            rolcanlogin;    /* allowed to log in as session user? */
        bool            rolreplication; /* role used for streaming replication */
        bool            rolbypassrls;   /* bypasses row level security? */
        bool            rolpolicyexempt; /* allowed to bypass VPD policies? */
        int32           rolconnlimit;   /* max connections allowed (-1=no limit) */
        Oid                     rolprofile;             /* EDB: OID of role's profile */
        int16           rolaccountstatus;       /* EDB: account status of the role */
        int32           rolfailedlogins;        /* EDB: number of faile

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+title: "PG_CLASS"
+---
+
+The `PG_CLASS` view provides information about tables and other objects that have columns or in other ways are similar to a table. This includes indexes, sequences, views, and other objects. In the context of this table "objects" refer to "relations." In addition to the information in the [PostgreSQL pg_class description](https://www.postgresql.org/docs/current/catalog-pg-class.html), EDB Postgres Advanced Server provides the following information:
+
+| Name                  | Type    | Description                                                                                           |
+| --------------------- | ------- | ----------------------------------------------------------------------------------------------------- |
+| relbackingtable       | "char"  | Specifies whether a table is backing relation for any other object. The default value is `d` for a normal object and `q` for queue tables. We store the relbackingtable as `q` for all the backing objects of QUEUE TABLE (backing table, index, toast table, and so forth). |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx:9]: Using relbackingtable we can identify whether table is backing relation for
 * any other object. Default value for relbackingtable will be 'd' for normal
 * object and 'q' for queue tables. We store the relbackingtable as 'q' for all
 * the backing object of QUEUE TABLE (backing table, index, toast table, etc).
 */

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+| relhasredactionpolicy | boolean | Object has a redaction policy  |
+| relpartname           | text    | Partition name |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx:11]: partition name

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+title: "PG_CLASS"
+---
+
+The `PG_CLASS` view provides information about tables and other objects that have columns or in other ways are similar to a table. This includes indexes, sequences, views, and other objects. In the context of this table "objects" refer to "relations." In addition to the information in the [PostgreSQL pg_class description](https://www.postgresql.org/docs/current/catalog-pg-class.html), EDB Postgres Advanced Server provides the following information:
+
+| Name                  | Type    | Description                                                                                           |
+| --------------------- | ------- | ----------------------------------------------------------------------------------------------------- |
+| relbackingtable       | "char"  | Specifies whether a table is backing relation for any other object. The default value is `d` for a normal object and `q` for queue tables. We store the relbackingtable as `q` for all the backing objects of QUEUE TABLE (backing table, index, toast table, and so forth). |
+| relhasredactionpolicy | boolean | Object has a redaction policy  |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_class.mdx:10]: has redaction policy

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_collation.mdx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+title: "PG_COLLATION"
+---
+
+The `PG_COLLATION` view describes the available collations, which map SQL names to operating system locale categories. In addition to the information in the [PostgreSQL pg_collation description](https://www.postgresql.org/docs/current/catalog-pg-collation.html), EDB Postgres Advanced Server provides the following information:
+
+| Name         | Type | Description                                                                                           |
+| ------------ | ---- | ----------------------------------------------------------------------------------------------------- |
+| collicu      | name | An ICU short form string, or empty. An ICU short form is a method of specifying collation attributes. |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_collation.mdx:9]: EDB: ICU short form string, or empty */

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
+title: "PG_NAMESPACE"
+---
+
+The `PG_NAMESPACE` view describes namespaces, which are the structures underlying SQL schemas. Each namespace can have a separate collection of relations, types, and so forth without name conflicts. In addition to the information in the [PostgreSQL pg_namespace description](https://www.postgresql.org/docs/current/catalog-pg-namespace.html), EDB Postgres Advanced Server provides the following information:
+
+| Name               | Type    | Description                          |
+| ------------------ | ------- | ------------------------------------ |
+| nspbodysource      | text    |                                      |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx:9]: i don't see this column in EPAS or PG.

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_synonym.mdx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+| cmin         |      |                                                         |
+| synlink      | text | Name of the database link, or an empty string           |                 
+| synname      | name | Name of the synonym                                     |
+| synnamespace | oid  | Namespace where the synonym resides                     |
+| synobjname   | text | Name of the base object referenced by the synonym       |
+| synobjschema | text | Name of the schema where the base object resides        |
+| synowner     | oid  | Object identifier of the owner (creator) of the synonym |
+| tableoid     |      |                                                         |              
+| xmax         |      |                                                         |   
+| xmin         |      |                                                         |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_synonym.mdx:19]: in EPAS v16, we have only these columns 

edb=# \d pg_synonym
            Table "pg_catalog.pg_synonym"
    Column    | Type | Collation | Nullable | Default 
--------------+------+-----------+----------+---------
 oid          | oid  |           | not null | 
 synname      | name |           | not null | 
 synnamespace | oid  |           | not null | 
 synowner     | oid  |           | not null | 
 synobjschema | text | C         |          | 
 synobjname   | text | C         | not null | 
 synlink      | text | C         |          | :

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx (hunk 8 above)
Comment (tusharahuja) [reply product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx:9]: These are the columns which are additionally added in EPAS 

nspparent          | oid       |           | not null | 
 nspobjecttype      | oid       |           | not null | 
 nspforeignserver   | oid       |           | not null | 
 nspsecdef          | boolean   |           | not null | 
 nspcompoundtrigger | boolean   |           | not null | 
 nspremoteschema    | text      | C         |          | 
 nspheadsrc         | text      | C         |          | 
 nspbodysrc

File: product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+| nspcompoundtrigger | boolean |                                      |
```
Comment (tusharahuja) [inline product_docs/docs/epas/16/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/pg_namespace.mdx:10]: nspparent= The OID of the parent table

**Comment by nidhibhammar** [comment]
@tusharahuja, can you have look at it, and let us know if any further changes are required

**Review by tusharahuja — COMMENTED** [review]
Screenshot 2024-01-05 at 12 22 54 PM

if we go to this page 
https://deploy-preview-5038--edb-docs-staging.netlify.app/docs/epas/latest/reference/oracle_compatibility_reference/epas_compat_cat_views/pg_views/

We are saying 
"
System catalog views v16
System catalog views are present in PostgreSQL. They may have extra information when accessed in EDB Postgres Advanced Server. The [PG_USER] view is an example of a system catalog view that provides additional information specific to EDB Postgres Advanced Server.
"
We  need to remove this line 

"The [PG_USER] view is an example of a system catalog view that provides additional information specific to EDB Postgres Advanced Server.
"
from there.

rest looks good.

**Comment by dwicinas** [comment]
@nidhibhammar I made the change suggested by Tushar. If you can mark it as reviewed, this one should be ready to merge.

**Review by nidhibhammar — APPROVED** [review]
LGTM

**Pull request merged**